edit/edit_test.go

```diff
@@ -251,79 +251,89 @@ func TestSubstitute(t *testing.T) {
 	tests := []subTest{
 		{
 			init: "Hello, 世界!",
-			addr: ",", re: ".*", sub: ``, g: true,
+			addr: ",", re: ".*", sub: ``, g: true, n: 1,
 			want: "", dot: addr{0, 0},
 		},
 		{
 			init: "Hello, 世界!",
-			addr: ",", re: "世界", sub: `World`,
+			addr: ",", re: "世界", sub: `World`, n: 1,
 			want: "Hello, World!", dot: addr{0, 13},
 		},
 		{
 			init: "Hello, 世界!",
-			addr: ",", re: "(.)", sub: `\1-`, g: true,
+			addr: ",", re: "(.)", sub: `\1-`, g: true, n: 1,
 			want: "H-e-l-l-o-,- -世-界-!-", dot: addr{0, 20},
 		},
 		{
 			init: "abcabc",
-			addr: ",", re: "abc", sub: "defg",
+			addr: ",", re: "abc", sub: "defg", n: 1,
 			want: "defgabc", dot: addr{0, 7},
 		},
 		{
 			init: "abcabcabc",
-			addr: ",", re: "abc", sub: "defg", g: true,
+			addr: ",", re: "abc", sub: "defg", g: true, n: 1,
 			want: "defgdefgdefg", dot: addr{0, 12},
 		},
 		{
 			init: "abcabcabc",
-			addr: "/abcabc/", re: "abc", sub: "defg", g: true,
+			addr: "/abcabc/", re: "abc", sub: "defg", g: true, n: 1,
 			want: "defgdefgabc", dot: addr{0, 8},
 		},
 		{
 			init: "abc abc",
-			addr: ",", re: "abc", sub: "defg",
+			addr: ",", re: "abc", sub: "defg", n: 1,
 			want: "defg abc", dot: addr{0, 8},
 		},
 		{
 			init: "abc abc",
-			addr: ",", re: "abc", sub: "defg", g: true,
+			addr: ",", re: "abc", sub: "defg", g: true, n: 1,
 			want: "defg defg", dot: addr{0, 9},
 		},
 		{
 			init: "abc abc abc",
-			addr: "/abc abc/", re: "abc", sub: "defg", g: true,
+			addr: "/abc abc/", re: "abc", sub: "defg", g: true, n: 1,
 			want: "defg defg abc", dot: addr{0, 9},
 		},
 		{
 			init: "abcabc",
-			addr: ",", re: "abc", sub: "de",
+			addr: ",", re: "abc", sub: "de", n: 1,
 			want: "deabc", dot: addr{0, 5},
 		},
 		{
 			init: "abcabcabc",
-			addr: ",", re: "abc", sub: "de", g: true,
+			addr: ",", re: "abc", sub: "de", g: true, n: 1,
 			want: "dedede", dot: addr{0, 6},
 		},
 		{
 			init: "abcabcabc",
-			addr: "/abcabc/", re: "abc", sub: "de", g: true,
+			addr: "/abcabc/", re: "abc", sub: "de", g: true, n: 1,
 			want: "dedeabc", dot: addr{0, 4},
 		},
 		{
 			init: "func f()",
-			addr: ",", re: `func (.*)\(\)`, sub: `func (T) \1()`, g: true,
+			addr: ",", re: `func (.*)\(\)`, sub: `func (T) \1()`, g: true, n: 1,
 			want: "func (T) f()", dot: addr{0, 12},
 		},
 		{
 			init: "abcdefghi",
-			addr: ",", re: "(abc)(def)(ghi)", sub: `\0 \3 \2 \1`,
+			addr: ",", re: "(abc)(def)(ghi)", sub: `\0 \3 \2 \1`, n: 1,
 			want: "abcdefghi ghi def abc", dot: addr{0, 21},
 		},
 		{
 			init: "abc",
-			addr: ",", re: "abc", sub: `\1`,
+			addr: ",", re: "abc", sub: `\1`, n: 1,
 			want: "", dot: addr{0, 0},
 		},
+		{
+			init: "abcabcabc",
+			addr: ",", re: "abc", sub: "def", n: 2,
+			want: "abcdefabc", dot: addr{0, 9},
+		},
+		{
+			init: "abcabcabc",
+			addr: ",", re: "abc", sub: "def", g: true, n: 2,
+			want: "abcdefdef", dot: addr{0, 9},
+		},
```

Comment:
We need more tests:
* `n==0`.
* `n==10` // multiple digit n.
* `n>1` and there are no occurrences.
* `n>1` but there is only 1 occurrence.
* maybe more.